core/src/main/java/com/datastrato/gravitino/catalog/TableOperationDispatcherWrapper.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.catalog;
+
+import com.datastrato.gravitino.EntityStore;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.exceptions.TableAlreadyExistsException;
+import com.datastrato.gravitino.listener.event.Event.UserAndIdentifierInfo;
+import com.datastrato.gravitino.listener.event.table.TableEvent.CreateTableEvent;
+import com.datastrato.gravitino.listener.event.table.TableEvent.DropTableEvent;
+import com.datastrato.gravitino.listener.event.table.TableExceptionEvent.CreateTableExceptionEvent;
+import com.datastrato.gravitino.listener.event.table.TableExceptionEvent.DropTableExceptionEvent;
+import com.datastrato.gravitino.listener.impl.EventBus;
+import com.datastrato.gravitino.listener.info.TableInfo;
+import com.datastrato.gravitino.rel.Column;
+import com.datastrato.gravitino.rel.Table;
+import com.datastrato.gravitino.rel.expressions.distributions.Distribution;
+import com.datastrato.gravitino.rel.expressions.sorts.SortOrder;
+import com.datastrato.gravitino.rel.expressions.transforms.Transform;
+import com.datastrato.gravitino.rel.indexes.Index;
+import com.datastrato.gravitino.storage.IdGenerator;
+import java.util.Map;
+
+public class TableOperationDispatcherWrapper extends TableOperationDispatcher {
```

Comment:
Maybe you can use wrapper pattern design to achieve this like here https://zhuanlan.zhihu.com/p/43368677

Reply:
done